src/recording/macromanager.cpp

```diff
@@ -0,0 +1,61 @@
+#include "macromanager.h"
+
+// TODO(xerus) handle track eject while recording
+// TODO(xerus) write tests
+// TODO(xerus) make recording button blink when state is armed
+
+MacroManager::MacroManager()
+        : m_COToggleRecording(ControlPushButton(ConfigKey(kMacroRecordingKey, "recording_toggle"))),
```

Comment:
As @uklotzde pointed out at Zulip, you're invoking the copy constructor here. You're first constructing a ControlPushbutton object, and then pass it as a parameter to the constructor for `m_COToggleRecording`.

Try:
```suggestion
        : m_COToggleRecording(ConfigKey(kMacroRecordingKey, "recording_toggle")),
```

Reply:
Ah thanks, another mistake due to the lack of my C++ knowledge :)